mux_test.go

```diff
@@ -1014,6 +1004,26 @@ func TestBuildVarsFunc(t *testing.T) {
 	}
 }
 
+// Addresses https://github.com/gorilla/mux/issues/200
+func TestCorrectVars(t *testing.T) {
+	tests := []routeTest{
+		{
+			title:        "Correct nested regex vars",
+			route:        new(Route).PathPrefix("/{id}/{kind:(a|b)}/{idx:[0-9]+}"),
```

Comment:
This pattern has a capturing group, which is not allowed. `(a|b)` should be `(?:a|b)` instead. When using `(?:a|b)` the test passes with the `extractVars` implementation of current master. This would mean to me that #200 is invalid, because it is caused by a bad pattern input. However, some automatic check with a nicer error message would definitely be helpful in such cases.